macos/library/FLEViewController.m

```diff
@@ -163,37 +170,49 @@ - (BOOL)addPlugin:(nonnull id<FLEPlugin>)plugin {
   return [self _addPlugin:plugin];
 }
 
+- (void)addKeyResponder:(nonnull NSResponder*)responder {
+  [self.additionalKeyResponders addObject:responder];
+}
+
+- (void)removeKeyResponder:(nonnull NSResponder*)responder {
+  [self.additionalKeyResponders removeObject:responder];
+}
+
 // TODO: Move to a structure that mimics the FlutterChannel structure used on iOS, for better
 // consistency (and to allow alternate method codecs).
 - (void)invokeMethod:(nonnull NSString *)method
            arguments:(nullable id)arguments
            onChannel:(nonnull NSString *)channel {
   NSDictionary *messageObject =
       [[[FLEMethodCall alloc] initWithMethodName:method arguments:arguments] asMessage];
-  if (![NSJSONSerialization isValidJSONObject:messageObject]) {
-    NSLog(@"Error: Unable to construct a valid JSON object from %@, %@", method, arguments);
-    return;
-  }
-
-  NSError *error = nil;
-  NSData *messageData =
-      [NSJSONSerialization dataWithJSONObject:messageObject options:0 error:&error];
-  if (!messageData) {
-    NSLog(@"Error: Failed to create JSON message data for %@: %@", method, error.debugDescription);
-    return;
-  }
-
-  FlutterPlatformMessage platformMessage = {
-      .struct_size = sizeof(FlutterPlatformMessage),
-      .channel = [channel UTF8String],
-      .message = messageData.bytes,
-      .message_size = messageData.length,
-  };
+  [self dispatchMessage:messageObject onChannel:channel];
+}
 
-  FlutterResult result = FlutterEngineSendPlatformMessage(_engine, &platformMessage);
-  if (result != kSuccess) {
-    NSLog(@"Unable to invoke %@ on Flutter engine (%d).", method, result);
-  }
+- (void)dispatchMessage:(nonnull NSDictionary*)message onChannel:(nonnull NSString *)channel {
+    if (![NSJSONSerialization isValidJSONObject:message]) {
+        NSLog(@"Error: Unable to construct a valid JSON object from %@", message);
+        return;
+    }
+    
+    NSError *error = nil;
+    NSData *messageData =
+    [NSJSONSerialization dataWithJSONObject:message options:0 error:&error];
+    if (!messageData) {
+        NSLog(@"Error: Failed to create JSON message data for %@: %@", message, error.debugDescription);
+        return;
+    }
+    
+    FlutterPlatformMessage platformMessage = {
+        .struct_size = sizeof(FlutterPlatformMessage),
+        .channel = [channel UTF8String],
+        .message = messageData.bytes,
+        .message_size = messageData.length,
+    };
+    
+    FlutterResult result = FlutterEngineSendPlatformMessage(_engine, &platformMessage);
+    if (result != kSuccess) {
+        NSLog(@"Flutter engine send unsuccessful response for message %@: (%d).", message, result);
```

Comment:
Typo: send->sent

Also, since the dictionary could be a lot to display, I'd put the error response before it:
Flutter engine sent unsuccessful response (%d) for message %@